eduHub/src/main/kotlin/com/arsiu/eduhub/nats/listeners/assignment/AssignmentNatsUpdateListener.kt

```diff
@@ -0,0 +1,42 @@
+package com.arsiu.eduhub.nats.listeners.assignment
+
+import com.arsiu.eduhub.nats.listeners.NatsListener
+import com.arsiu.eduhub.protobuf.handlers.assignment.FindAllStreamHandler
+import com.arsiu.eduhub.shared.streams.SharedAssignmentStream
+import com.arsiu.eduhub.v2.assignmentsvc.NatsSubject.ASSIGNMENT_UPDATE_BUS
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentProto
+import com.google.protobuf.Parser
+import io.nats.client.Connection
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+
+@Component
+class AssignmentNatsUpdateListener(
```

Comment:
what if you run 2 instances of your app, will each of them handle the update?

Reply:
Both will notify their clients(which sub to stream)that some entity was updated.
Tested even got screenshots.

![image](https://github.com/ArSiu/eduHub/assets/33807305/6b44531a-8c60-45d5-9a4e-e53ebeb59b62)
![image](https://github.com/ArSiu/eduHub/assets/33807305/01a4bc5c-a68e-4975-a07f-43f964a5d4ca)
![image](https://github.com/ArSiu/eduHub/assets/33807305/02a18eff-f155-4886-af98-949149d68325)
![image](https://github.com/ArSiu/eduHub/assets/33807305/151e011f-5cb9-406e-a082-3c805bb6d0b4)

there are two obj updates named "ddddd" because that was two separate updates

Maybe I don’t see issue or misunderstood task so pls give more details.

Reply:
I expected randomQueue to ensure that each instance will handle update, but you're subscribing without queue which works the same, missed this